docs/DeveloperGuide.md

```diff
@@ -1,34 +1,972 @@
+Based on the type of command from the user, `parseCommand()` then calls different methods to extract parameters and 
+flags from the user command that are relevant to the command. The parser then creates 
+the relevant `Command` object and dispatches the control of the program to the created object. 
+
+> ℹ️ **Note:** Interpretation and checking of parameter validity occurs in the parser. 
+
+The methods that extract various components include
+- `extractModuleCode()` - Extracts the module code by checking for the parameter `-c` flag. Invoked with `createAddCommand()`. 
+- `extractModuleType()` - Extracts the module type by checking for the parameter `-t` flag. Invoked with `createAddCommand()`. 
+- `extractModuleCredits()` - Extracts the module type by checking for the parameter `-mc` flag. Invoked with `createAddCommand()`, `createUpdateCommand()`. 
+- `extractModuleGrade()` - Extracts the grade obtained by checking for the parameter `-g` flag. Invoked with `createDoneCommand()`, `createUpdateCommand()`. 
+- `extractModuleName()` - Extracts the module name by checking for the parameter `-n` flag. Invoked with `createUpdateCommand()`
+- `extractPreRequisites()` - Extracts the prerequisites of the module by checking for the `-p` flag. Invoked with `createAddCommand()`, `createUpdateCommand()`. 
+- `extractListScope()` - Extracts the list type. Accepts all, complete,  incomplete or available as acceptable scopes. Invoked with `createListCommand()`. 
+
+which would be invoked based on the command word detected in the user input. 
+
+> ℹ️ **Note:** The Parser also contains checks that ensures the parameters passed are appropriate
+
+The methods that check various parameters include
+- `isModuleCodeValid` - Checks that the module code is a valid module code according to the standard for NUS modules. The method uses regex to check for the valid code. There are 2 overloading methods, one for checking a single instance and another for an arraylist to check through a list of module codes. 
+
+### 3.3.2 Command
+
+#### Description
+The `command` component executes the correct command based on what the parser interprets.
+
+#### Design
+The `command` component consists of an abstract class `Command` and 8 subclasses that inherit from it.
+
+The 8 subclasses are:
+* AddCommand
+* CapCommand
+* DeleteCommand
+* DoneCommand
+* ExitCommand
+* ListCommand
+* ProgressCommand
+* UpdateCommand
+
+The correct command is executed once the `Command` object is created by the parser by executing the `execute()` method in the correct subclass.
+The command execution can affect the `Model` (eg. adding a module).
+At the end of each command execution, different methods in the `Ui` will be called to perform certain actions, such as displaying the list of modules to the user.
+
+Below is a Command class diagram.
+
+![archi](images/CommandClassDiagram.png)
+
+<sup>***Figure 3.3.2.1** UML class diagram for Command class*</sup>
```

Comment:
![image](https://user-images.githubusercontent.com/47781738/113468602-b1ceb600-9479-11eb-9794-d5b9b8d36917.png)
Is this the intended size for the diagram? Do you think it's better to split up the diagram into several reference diagram so that developers will have an easier time reading them?